config.go

```diff
@@ -166,7 +416,71 @@ func loadLndConfig(preCfg *Config) (*lnd.Config, error) {
 		log.Warnf("%v", configFileError)
 	}
 
-	return cleanCfg, nil
+	return cfg, nil
+}
+
+// validateRemoteModeConfig validates the terminal's own configuration
+// parameters that are only used in the "remote" lnd mode.
+func validateRemoteModeConfig(cfg *Config) error {
+	r := cfg.Remote
+
+	// Validate the network as in the remote node it's provided as a string
+	// instead of a series of boolean flags.
+	if _, err := lndclient.Network(r.Lnd.Network).ChainParams(); err != nil {
+		return fmt.Errorf("error validating lnd remote network: %v", err)
+	}
+	cfg.network = r.Lnd.Network
+
+	// If the remote lnd's network isn't the default, we also check if we
+	// need to adjust the default macaroon directory so the user can only
+	// specify --network=testnet for example if everything else is using
+	// the defaults.
+	if r.Lnd.Network != defaultNetwork &&
+		r.Lnd.MacaroonDir == defaultRemoteLndMacDir {
+
+		r.Lnd.MacaroonDir = filepath.Join(
+			lndDefaultConfig.DataDir, defaultLndChainSubDir,
+			defaultLndChain, r.Lnd.Network,
+		)
+	}
+
+	// If the provided lit directory is not the default, we'll modify the
+	// path to all of the files and directories that will live within it.
+	litDir := lnd.CleanAndExpandPath(cfg.LitDir)
+	if litDir != defaultLitDir {
+		r.LitTLSCertPath = filepath.Join(litDir, defaultTLSCertFilename)
+		r.LitTLSKeyPath = filepath.Join(litDir, defaultTLSKeyFilename)
+		r.LitLogDir = filepath.Join(litDir, defaultLogDirname)
+	}
+
+	r.LitTLSCertPath = lncfg.CleanAndExpandPath(r.LitTLSCertPath)
+	r.LitTLSKeyPath = lncfg.CleanAndExpandPath(r.LitTLSKeyPath)
+	r.LitLogDir = lncfg.CleanAndExpandPath(r.LitLogDir)
+
+	// Make sure the parent directories of our certificate files exist. We
+	// don't need to do the same for the log dir as the log rotator will do
+	// just that.
+	if err := makeDirectories(filepath.Dir(r.LitTLSCertPath)); err != nil {
+		return err
+	}
+	if err := makeDirectories(filepath.Dir(r.LitTLSKeyPath)); err != nil {
+		return err
+	}
+
+	logWriter := build.NewRotatingLogWriter()
+	cfg.Lnd.LogWriter = logWriter
+	err := logWriter.InitLogRotator(
```

Comment:
shouldn't this also be set up in integrated mode?

Reply:
In integrated mode, we call `lnd.ValidateConfig()` which creates the default log writer. Added a comment to clarify this here.